pilot/pkg/config/kube/ingress/conversion.go

```diff
@@ -105,22 +106,49 @@ func ConvertIngressV1alpha3(ingress v1beta1.Ingress, domainSuffix string) (model
 		Hosts: []string{"*"},
 	})
 
-	virtualService := &networking.VirtualService{
-		Hosts:    []string{"*"},
-		Gateways: []string{model.IstioIngressGatewayName},
+	gatewayConfig := model.Config{
+		ConfigMeta: model.ConfigMeta{
+			Type:      model.Gateway.Type,
+			Group:     model.Gateway.Group,
+			Version:   model.Gateway.Version,
+			Name:      ingress.Name + "-" + model.IstioIngressGatewayName,
+			Namespace: model.IstioIngressNamespace,
+			Domain:    domainSuffix,
+		},
+		Spec: gateway,
 	}
 
-	var httpRoutes []*networking.HTTPRoute
+	return gatewayConfig
+}
+
+func ConvertIngressVirtualService(ingress v1beta1.Ingress, domainSuffix string, ingressByHost map[string]*model.Config) {
+	// Ingress allows a single host - if missing '*' is assumed
```

Comment:
public functions should be documented. i suggest moving much of the text like "mixing *" to the function comment since it will explain the behavior of the function to the reader without needing to go into the code.

Reply:
done.